drivers/onewire/onewire.c

```diff
@@ -63,41 +63,41 @@ static void _release(const onewire_t *owi)
     gpio_init(owi->pin, owi->imode);
 }
 
-static void _spin_us(uint32_t delay_us)
+static void _spin_us(uint32_t *t_ref, uint32_t delay_us)
 {
-    uint32_t target = xtimer_now_usec() + delay_us;
-    while (xtimer_now_usec() < target) {}
+    *t_ref = *t_ref + delay_us;
+    while (xtimer_now_usec() < *t_ref) {}
```

Comment:
when `t_ref + delay_us` overflows (e.g., at end of period), this will not wait at all.
Maybe restrict the delay_us parameter to uint16_t, then use sth like:

    while(((int32_t)((*t_ref + delay_us) - xtimer_now_usec())) > 0) { }

Reply:
yes, absolutely right. Will fix